lib/src/util.dart

```diff
@@ -96,6 +98,52 @@ String normalizeLinkLabel(String label) {
   return text;
 }
 
+///  Decodes HTML entity and numeric character references, for example decode
+/// `&#35` to `#`.
+String decodeHtmlCharacters(String input) =>
+    input.replaceAllMapped(htmlCharactersPattern, decodeHtmlCharacterFromMatch);
+
+/// Decodes HTML entity and numeric character references from the given [match].
+String decodeHtmlCharacterFromMatch(Match match) {
+  final text = match.match;
+  final isEntity = match[1] != null;
+  final isDecimal = match[2] != null;
+  final isHexadecimal = match[3] != null;
+  String? decodedText;
+
+  // Entity references, see
+  // https://spec.commonmark.org/0.30/#entity-references.
+  if (isEntity) {
+    decodedText = htmlEntitiesMap[text];
+  }
+
+  // Decimal numeric character references, see
+  // https://spec.commonmark.org/0.30/#decimal-numeric-character-references.
+  else if (isDecimal) {
+    final decimalValue = int.parse(match[2]!);
+    int hexValue;
+    if (decimalValue < 1114112 && decimalValue > 1) {
+      hexValue = int.parse(decimalValue.toRadixString(16), radix: 16);
+    } else {
+      hexValue = 0xFFFd;
+    }
+
+    decodedText = String.fromCharCode(hexValue);
```

Comment:
I think it would be simpler to just return this value. `return String.fromCharCode(hexValue);`.

Reply:
Fixed